src/System/src/FloatingBaseSystemDynamics.cpp

```diff
@@ -0,0 +1,182 @@
+    if (!m_kinDyn->getFreeFloatingMassMatrix(m_massMatrix))
+    {
+        std::cerr << "[FloatingBaseDynamicalSystem::dynamics] Unable to get the mass matrix."
+                  << std::endl;
+        return false;
+    }
+
+    // compute the generalized bias forces
+    if (!m_kinDyn->generalizedBiasForces(m_generalizedBiasForces))
+    {
+        std::cerr << "[FloatingBaseDynamicalSystem::dynamics] Unable to get the bias forces."
+                  << std::endl;
+        return false;
+    }
+
+    // here we want to compute the robot acceleration as
+    // robotAcceleration = M^-1 (-h + J' F + B tau) = M^-1 * m_knownCoefficent
+
+    // add the generalized bias forces to the known coefficent
+    iDynTree::toEigen(m_knownCoefficent).head<m_baseDoFs>()
+        = -iDynTree::toEigen(m_generalizedBiasForces.baseWrench());
+    iDynTree::toEigen(m_knownCoefficent).tail(m_actuatedDoFs)
+        = -iDynTree::toEigen(m_generalizedBiasForces.jointTorques());
+
+    // add the contact wrench to the knownCoefficent
+    for (const auto& contactWrench : contactWrenches)
+    {
+        // compute the contact jacobian
+        if (!m_kinDyn->getFrameFreeFloatingJacobian(contactWrench.index(), m_jacobianMatrix))
+        {
+            std::cerr << "[FloatingBaseDynamicalSystem::dynamics] Unable to get the Jacobian for "
+                         "the frame named: "
+                      << m_kinDyn->model().getFrameLink(contactWrench.index()) << "." << std::endl;
+            return false;
+        }
+
+        // update the state of the contact model
+        auto contactPtr = contactWrench.contactModel().lock();
+        if (contactPtr == nullptr)
+        {
+            std::cerr << "[FloatingBaseDynamicalSystem::dynamics] The contact model associated to "
+                         "the frame named: "
+                      << m_kinDyn->model().getFrameLink(contactWrench.index())
+                      << " has been expired." << std::endl;
+            return false;
+        }
+
+        contactPtr->setState(m_kinDyn->getFrameVel(contactWrench.index()),
+                             m_kinDyn->getWorldTransform(contactWrench.index()));
+
+        iDynTree::toEigen(m_knownCoefficent) += iDynTree::toEigen(m_jacobianMatrix).transpose()
+            * iDynTree::toEigen(contactPtr->getContactWrench());
+    }
+
+    // add the joint torques to the known coefficent
+    iDynTree::toEigen(m_knownCoefficent).tail(m_actuatedDoFs) += iDynTree::toEigen(jointTorques);
+
+    // resize the joint acceleration
+    jointAcceleration.resize(m_actuatedDoFs);
+
+    // compute the generalized robot acceleration solving the linear system. Here we assume that the
+    // mass matrix is positive definite (check here for further informations:
+    // https://eigen.tuxfamily.org/dox/group__TutorialLinearAlgebra.html)
+    iDynTree::toEigen(m_generalizedRobotAcceleration)
+        = iDynTree::toEigen(m_massMatrix).llt().solve(iDynTree::toEigen(m_knownCoefficent));
```

Comment:
Consider keeping the door open for regularization terms and/or reflected inertias.

Reply:
Done in ac3f5e3d45a577ae242b057ec6f8952cf6b74110